### ISSUE TITLE
Classic: SAPHanaController hangs with WAITING4NODES even when all nodes are up and running

### ISSUE THREAD

**Issue by fmherschel** [SUSE/SAPHanaSR-ScaleOut#103]
SAPHanaSR-showAttr shows that all role attributes for the nodes are set, but SAPHanaController is not able to start the instance, because it claims the nodes are missing.
It seams the detection of the worker roles was broken by code lines like:
read nNsConf nNsCurr nIsConf nIsCurr < <(echo ${nRole//:/ })

If the attributes is "master1::worker:" then the string "worker" is assigned to nNsCurr but not (as attended) to nIsConf.

Fix will follow...

**Comment by angelabriel** [comment]
good catch.
I was under the impression that this change was equivalent to the logic before
  ```
nRole=$(get_hana_attribute ${NODENAME} "${ATTR_NAME_HANA_ROLES[@]}" | tr ':' ' ')
  read nNsConf nNsCurr nIsConf nIsCurr <<< $nRole
```
But this change was introduced in 2020, so it was working for the last 3 years.
Luck or is their really a difference between `tr ':' ' ' ` and `${nRole//:/ }`?

**Comment by fmherschel** [comment]
It might be also a difference between the complete lines (so also < <(...) )
At least in our debug we did see that SAPHanaController ALWAYS got number of worker node is zero with the new code.

**Comment by fmherschel** [comment]
Same fix needed in SAPHanaTopology.
Fix follows.

**Comment by fmherschel** [comment]
Did we had something like "master1:-:worker:-" in the past?

**Comment by fmherschel** [comment]
Hopefully fixed in PR #104 and PR #105

**Comment by fmherschel** [comment]
Seams to be solved.
